_posts/2025-08-28-quarkus-a2a-java-0-3-0-alpha-release.adoc

```diff
@@ -0,0 +1,242 @@
+                .name("Weather Agent")
+                .description("Helps with weather")
+                .url("http://localhost:10001") <1>
+                .preferredTransport(TransportProtocol.JSONRPC.asString()) <2>
+                .additionalInterfaces(List.of( <3>
+                        new AgentInterface(TransportProtocol.JSONRPC.asString(), "http://localhost:10001"), <4>
+                        new AgentInterface(TransportProtocol.GRPC.asString(), "localhost:9000") <5>
+                ))
+                .version("1.0.0")
+                .capabilities(new AgentCapabilities.Builder()
+                        .streaming(true)
+                        .pushNotifications(false)
+                        .stateTransitionHistory(false)
+                        .build())
+                .defaultInputModes(Collections.singletonList("text"))
+                .defaultOutputModes(Collections.singletonList("text"))
+                .skills(Collections.singletonList(new AgentSkill.Builder()
+                        .id("weather_search")
+                        .name("Search weather")
+                        .description("Helps with weather in city, or states")
+                        .tags(Collections.singletonList("weather"))
+                        .examples(List.of("weather in LA, CA"))
+                        .build()))
+                .protocolVersion("0.3.0")
+                .build();
+    }
+}
+----
+<1> The primary endpoint URL for our A2A server agent, `http://localhost:10001` in this example.
+<2> The preferred transport for our A2A server agent, `JSON-RPC` in this example. This is the transport protocol available at the primary endpoint URL.
+<3> The optional additional interfaces supported by our A2A server agent. This should include all supported transports.
+<4> The primary endpoint URL should also be specified here for completeness.
+<5> The gRPC transport URL, `localhost:9000`, indicating that our A2A server agent also supports gRPC.
+
+=== A2A Client Updates
+
+To make use of an A2A client using the A2A Java SDK, add the following dependency to your project:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.github.a2asdk</groupId>
+    <artifactId>a2a-java-sdk-client</artifactId> <1>
+</dependency>
+----
+<1> The `a2a-java-sdk-client` dependency provides access to a `ClientBuilder` that you can use to create your A2A `Client`.
+
+==== Supported Transports
+
+The A2A Java SDK client implementation now supports the gRPC transport protocol in addition to JSON-RPC.
+
+As mentioned previously, support for the HTTP+JSON/REST transport will be coming soon!
+
+===== JSON-RPC 2.0 Transport
+
+To allow your client to support the JSON-RPC transport for communication, just `a2a-java-sdk-client` is needed. No additional dependencies need to be manually added to your project.
+
+===== gRPC Transport
+
+To allow your client to support the gRPC transport for communication, simply add the following dependency to your project:
+
```

Comment:
For both the client and server, does it use Quarkus legacy gRPC or Quarkus "vertx-based" gRPC? 
As I see port 9000, I believe it's the legacy server.

Reply:
@cescoffier Thanks for pointing this out, I've created https://github.com/a2aproject/a2a-java/issues/250 to track setting this property in our test configuration.

Reply:
I've also updated the code snippet in the blog post for this.